qiskit/transpiler/_transpiler.py

```diff
@@ -139,22 +136,37 @@ def _transpile_dags(dags, basis_gates='u1,u2,u3,cx,id', coupling_map=None,
     Raises:
         TranspilerError: if the format is not valid.
     """
-    # TODO: Parallelize this method
-    final_dags = []
-    for dag, initial_layout in zip(dags, initial_layouts):
-        final_dag, final_layout = transpile(
-            dag,
-            basis_gates=basis_gates,
-            coupling_map=coupling_map,
-            initial_layout=initial_layout,
-            get_layout=True,
-            seed=seed,
-            pass_manager=pass_manager)
-        final_dag.layout = [[k, v] for k, v in final_layout.items()] if final_layout else None
-        final_dags.append(final_dag)
+
+    values = list(zip(dags, initial_layouts))
+    final_dags = parallel_map(_transpile_dags_parallel, values,
+                              task_kwargs={'basis_gates': basis_gates,
+                                           'coupling_map': coupling_map,
+                                           'get_layout': True,
+                                           'seed': seed,
+                                           'pass_manager': pass_manager})
     return final_dags
 
 
+def _transpile_dags_parallel(value, **kwargs):
```

Comment:
1) Ok
2) Ok
3) This is because there is only a single iteration variable once the terms are grouped into a tuple, and the pair are necessarily needed together.

Reply:
> This is because there is only a single iteration variable once the terms are grouped into a tuple, and the pair are necessarily needed together.

Hmm, but then we might have a design problem bigger than expected - basically, with your proposal, we are enforcing that the functions passed to `parallel_map` always need to have at maximum one parameter that is the one that will differ between each task, right? I'll move the discussion to the main thread in the PR, as I think this is an artificial constraint that might need some extra thinking.